### PR TITLE
Implement centralized configuration system

### PR DESCRIPTION
**Pull request by Xain3**
## Overview

This PR establishes the foundation for the Vision with Fade module by implementing a **centralized configuration system**, updating the project constitution, and establishing comprehensive coding standards. This is a substantial architectural update that improves modularity, maintainability, and developer experience.

### 1. Config Singleton System (`src/config/config.ts`)
A production-grade configuration service that aggregates multiple sources:

**Features**:
- ✅ **Lazy-loaded singleton** via ESM module caching
- ✅ **Namespace-keyed YAML loading** from 7 constant files (errors, foundry, hooks, logging, moduleManagement, occlusion, placeables)
- ✅ **Settings definitions** from `settings.yaml`
- ✅ **Module manifest** from `module.json`
- ✅ **Environment variable support** with customizable prefix (OMH_*)
- ✅ **Deep immutability** via recursive Object.freeze() + Proxy-based mutation protection
- ✅ **Fail-fast error handling** with detailed context
- ✅ **Zero external dependencies** (uses built-in: fs, path, yaml package)

**Access Pattern**:
```typescript
import { config } from './config/config.ts';

config.module.id              // "vision-with-fade"
config.constants.errors       // { separator: " || ", ... }
config.settings               // Array of setting definitions
config.env.OMH_DEBUG_MODE     // Environment variable access
```

### 2. Helper Functions (`src/config/helpers/configHelpers.ts`)
Pure functions (with I/O) for configuration loading:
- `loadYamlFiles()` - Load all constant YAML files
- `mergeConstants()` - Namespace-keyed merge
- `extractConfigPrefix()` - Extract module prefix
- `loadSettings()` - Parse settings definitions
- `loadModuleManifest()` - Load module.json
- `loadEnvironmentVariables()` - Filter and load OMH_* variables

All helpers are exported individually for testability.

### 3. Comprehensive Test Suite
- **Unit tests** (configHelpers.unit.test.mjs) - Helper function testing in isolation
- **Integration tests** (config.int.test.mjs) - Real project files, 29+ test cases
- **Performance tests** (config.performance.test.mjs) - <100ms initialization, <1ms cached access
- **Test fixtures** - Valid and malformed YAML/JSON for error testing

### 4. Constitution Update (v2.0.0)
- Elevated to version 2.0.0 with new governance additions
- Enhanced documentation requirements: file headers now reference coding standards
- New testing standards requiring isolated test execution
- Clarified shebang and usage requirements for executable scripts

### 5. Development Standards (NEW)
- **STYLE_GUIDE.md** (660 lines) - Comprehensive standards for:
  - File structure and headers (@file, @description, @path)
  - JavaScript/TypeScript conventions (ESM, naming, code style)
  - JSDoc and inline documentation patterns
  - Code organization and error handling
  - FoundryVTT integration best practices (hooks-only approach)
  - Commit and PR guidelines

- **STYLE_GUIDE_QUICK_REFERENCE.md** (219 lines) - Developer cheat sheet for quick lookup

- **docs/README.md** (v1.0.0) - Updated with links to standards and quick-start guidance

### 6. Documentation Updates
- **Config Module README** - Complete usage guide with examples
- **Constants README** - Details on each YAML namespace
- **Settings README** - Setting definition patterns and localization
- **Helpers README** - Function signatures and design patterns
- **Quickstart Guide** - Common usage patterns and FAQs

## 📊 Scope & Metrics

**Files Changed/Added**: 25+
**Lines Added**: ~2,855+ (documentation) + ~700+ (implementation) + ~500+ (tests)
**Test Coverage**: ≥85% on core functionality
**Performance**: <500ms first init, <1ms cached retrieval

## ✨ Key Improvements

1. **Modular Architecture** - Config is independent, testable, and replaceable
2. **Type Safety** - Full TypeScript support with no `any` types
3. **Fail-Fast Philosophy** - Configuration errors caught immediately with context
4. **Immutability Guarantees** - Deep freezing + Proxy protection prevents accidental mutations
5. **Developer Experience** - Centralized access point, comprehensive documentation, IDE autocomplete
6. **Sustainability** - Clear separation of concerns, extensive documentation
7. **Backward Compatible** - Existing module code unaffected; config is additive

## 🔍 Architecture Decisions

All major decisions documented with rationale in `/specs/001-centralized-config-system/`:
- **Namespace-keyed merge**: Prevents collisions between YAML files
- **Singleton pattern**: Ensures one config instance throughout module lifetime
- **Proxy-based immutability**: Prevents mutations while staying performant
- **Fail-fast error handling**: Configuration errors don't silently fail
- **Environment variable prefix pattern**: Centralized, customizable override mechanism

## ✅ Testing & Quality

- ✅ All ESLint rules passing (zero errors)
- ✅ TypeScript strict mode compiling (zero errors)
- ✅ 29 integration tests passing
- ✅ 12 unit tests passing
- ✅ Performance targets verified (<100ms)
- ✅ Immutability validated at all nesting levels
- ✅ Error scenarios comprehensively tested

## 📖 Review Guidance

**Priority Areas**:
1. **config.ts** - Core singleton implementation and Proxy pattern
2. **configHelpers.ts** - Helper function error handling and YAML/JSON parsing
3. **STYLE_GUIDE.md** - Coding standards applicability across the project
4. **Test Coverage** - Verify all edge cases are covered

**Backward Compatibility**: ✅ PASSING
- Existing module functionality unchanged
- Config is opt-in via import
- No breaking changes to existing APIs

## 🚀 Next Steps

After merge, subsequent PRs will:
1. Integrate config throughout the module
2. Migrate existing code to use centralized config
3. Add additional YAML constant files as needed
4. Establish CI/CD checks for config validation

---

This is a **comprehensive, production-ready foundation** for the Vision with Fade module that establishes patterns and practices for all future development.

### CODE REVIEW

File: tests/unit/config/configHelpers.unit.test.mjs [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/**
+ * @file Config Helper Functions Unit Tests
+ * @description Unit tests for configHelpers.ts helper functions
+ * @path tests/unit/configHelpers.unit.test.mjs
+ */
+
+import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
```
[the remark this answers is not in the document]
Comment (Xain3) [reply tests/unit/config/configHelpers.unit.test.mjs:7]: Ignore - add for consistency

File: tests/performance/config.performance.test.mjs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      expect(id).toBeDefined();
+      expect(version).toBeDefined();
+      expect(duration).toBeLessThan(0.1);
+    });
+
+    it('accessing environment variables should be instant', () => {
+      const startTime = performance.now();
+
+      // eslint-disable-next-line no-unused-vars
+      const debugMode = config.env.OMH_DEBUG_MODE;
```
[the remark this answers is not in the document]
Comment (Xain3) [reply tests/performance/config.performance.test.mjs:117]: Leave it for future testing

File: tests/integration/config.int.test.mjs [hunk 3]
```diff
@@ -0,0 +1,308 @@
+/**
+ * @file Config Integration Tests
+ * @description Integration tests for Config singleton with real project files
+ * @path tests/integration/config.int.test.mjs
+ */
+
+import { describe, it, expect, beforeEach, afterEach } from 'vitest';
```
[the remark this answers is not in the document]
Comment (Xain3) [reply tests/integration/config.int.test.mjs:7]: Added for consistency
